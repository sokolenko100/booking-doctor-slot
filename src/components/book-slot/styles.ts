import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  bookSlotButton: {
    width: '100%',
    flexDirection: 'row',
    alignItems: 'center',
    padding: 10,
    backgroundColor: '#f8f9fa',
    flex: 1,
    height: 45,
    borderRadius: 5,
    borderColor: '#bcbcbc',
    borderWidth: 0.2,
  },
  bookSlotText: {
    color: '#007BFF',
    fontSize: 14,
    marginLeft: 5,
    fontWeight: '600',
  },
});
