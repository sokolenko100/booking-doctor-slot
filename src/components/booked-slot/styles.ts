import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  bookedSlot: {
    flex: 1,
    height: 50,
    padding: 7,
    borderRadius: 3,
    borderLeftWidth: 3,
    backgroundColor: '#d1ecf1',
    borderLeftColor: '#00ff00',
  },
  bookedText: {
    fontSize: 16,
    fontWeight: '600',
    color: '#00ff00',
  },
});
