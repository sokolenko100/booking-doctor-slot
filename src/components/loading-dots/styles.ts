import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    height: 50,
  },
  dot: {
    width: 10,
    height: 10,
    borderRadius: 15,
    backgroundColor: '#007BFF',
    marginHorizontal: 5,
  },
});
