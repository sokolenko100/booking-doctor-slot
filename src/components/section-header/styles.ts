import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  container: {
    backgroundColor: '#FFFFFF',
    marginBottom: 10,
    height: 40,
    paddingHorizontal: 10,
    justifyContent: 'center',
    shadowColor: 'black',
    shadowOffset: {width: 0, height: 1},
    shadowOpacity: 0.2,
    shadowRadius: 3,
    elevation: 2,
  },
  sectionHeader: {
    fontSize: 16,
    fontWeight: '500',
    color: '#999999',
  },
});
