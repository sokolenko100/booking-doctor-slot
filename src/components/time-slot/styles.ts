import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    marginVertical: 5,
    marginRight: 10,
  },
  timeContainer: {
    marginHorizontal: 10,
    marginRight: 20,
    alignItems: 'center',
  },
  time: {
    fontSize: 16,
    color: '#999999',
    fontWeight: '500',
    width: 47,
    marginLeft: 3,
  },
});
