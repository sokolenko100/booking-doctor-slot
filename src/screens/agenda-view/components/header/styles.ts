import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    padding: 15,
    backgroundColor: '#ffffff',
  },
  dateRange: {
    fontSize: 16,
    fontWeight: '500',
    color: '#999999',
  },
});
