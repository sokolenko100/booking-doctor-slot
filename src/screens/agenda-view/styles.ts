import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    backgroundColor: '#ffffff',
  },
  container: {
    flex: 1,
    backgroundColor: '#eeeeee',
  },
  buttonContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 10,
  },
  header: {
    fontSize: 18,
    fontWeight: 'bold',
    marginVertical: 10,
  },
  loadingContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
