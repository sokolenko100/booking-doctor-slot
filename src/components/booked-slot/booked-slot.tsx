import React, {FC} from 'react';
import {Text, View} from 'react-native';
import {styles} from './styles';

interface IBookedSlot {
  color?: string;
}

/**
 * A functional component representing a booked slot in a calendar or schedule.
 * It displays a view with a border and a text indicating that the slot is booked.
 *
 * @param {string} [color] - Optional color for the border and text. Defaults to black.
 * @returns {JSX.Element} - A React component representing a booked slot.
 */
export const BookedSlot: FC<IBookedSlot> = ({color}) => {
  return (
    <View
      testID={'bookedSlotID'}
      style={[styles.bookedSlot, {borderLeftColor: color}]}>
      <Text style={[styles.bookedText, {color: color}]}>Slot booked</Text>
    </View>
  );
};
