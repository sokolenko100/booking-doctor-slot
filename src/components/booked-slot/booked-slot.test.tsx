import {render, waitFor} from '@testing-library/react-native';
import React from 'react';
import {BookedSlot} from './booked-slot';
import {styles} from './styles';

describe('BookedSlot component', () => {
  it('renders the correct text', async () => {
    const {getByText} = render(<BookedSlot color={'red'} />);
    const text = await waitFor(() => getByText('Slot booked'));
    expect(text).toBeDefined();
  });

  it('renders with default border left color when no color is provided', async () => {
    const {getByTestId} = render(<BookedSlot />);
    const slot = await waitFor(() => getByTestId('bookedSlotID'));
    expect(slot.props.style[0].borderLeftColor).toBe(
      styles.bookedSlot.borderLeftColor,
    );
  });

  it('renders with correct text color when color is provided', async () => {
    const {getByText} = render(<BookedSlot color={'red'} />);
    const text = await waitFor(() => getByText('Slot booked'));
    expect(text.props.style[1].color).toBe('red');
  });

  it('renders with default text color when no color is provided', async () => {
    const {getByText} = render(<BookedSlot />);
    const text = await waitFor(() => getByText('Slot booked'));
    expect(text.props.style[0].color).toBe(styles.bookedText.color);
  });

  it('matches the snapshot', async () => {
    const output = render(<BookedSlot />);

    expect(output).toMatchSnapshot();
  });
});
