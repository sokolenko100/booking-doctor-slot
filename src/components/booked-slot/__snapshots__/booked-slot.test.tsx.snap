// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookedSlot component matches the snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#d1ecf1",
        "borderLeftColor": "#00ff00",
        "borderLeftWidth": 3,
        "borderRadius": 3,
        "flex": 1,
        "height": 50,
        "padding": 7,
      },
      {
        "borderLeftColor": undefined,
      },
    ]
  }
  testID="bookedSlotID"
>
  <Text
    style={
      [
        {
          "color": "#00ff00",
          "fontSize": 16,
          "fontWeight": "600",
        },
        {
          "color": undefined,
        },
      ]
    }
  >
    Slot booked
  </Text>
</View>
`;
